File: add/codesnippet/CPP/p-system.net.sockets.soc_0_1.cpp
   client->Connect( anEndPoint );
   if (  !client->Connected )
   {
      Console::WriteLine( "Winsock error: {0}", Convert::ToString(
         System::Runtime::InteropServices::Marshal::GetLastWin32Error() ) );
   }
      
   // This is how you can determine whether a socket is still connected.
   bool blockingState = client->Blocking;
   try
   {
      array<Byte>^tmp = gcnew array<Byte>(1);
      client->Blocking = false;
      client->Send( tmp, 0, static_cast<SocketFlags>(0) );
      Console::WriteLine( L"Connected!" );
   }
   catch ( SocketException^ e ) 
   {
      // 10035 == WSAEWOULDBLOCK
      if ( e->NativeErrorCode.Equals( 10035 ) )
      {
         Console::WriteLine( "Connected from an exception!" );
      }
      else
      {
         Console::WriteLine( "Disconnected: {0}!", e->NativeErrorCode );
      }
   }
   finally
   {
      client->Blocking = blockingState;
   }

   Console::WriteLine( "Connected: {0}", client->Connected );

File: add/codesnippet/VisualBasic/p-system.net.sockets.soc_0_1.vb
        ' .Connect throws an exception if unsuccessful
        client.Connect(anEndPoint)
        
        ' This is how you can determine whether a socket is still connected.
        Dim blockingState As Boolean = client.Blocking
        Try
            Dim tmp(0) As Byte
            
            client.Blocking = False
            client.Send(tmp, 0, 0)
            Console.WriteLine("Connected!")
        Catch e As SocketException
            ' 10035 == WSAEWOULDBLOCK
            If e.NativeErrorCode.Equals(10035) Then
                Console.WriteLine("Still Connected, but the Send would block")
            Else
                Console.WriteLine("Disconnected: error code {0}!", e.NativeErrorCode)
            End If
        Finally
            client.Blocking = blockingState
        End Try
        
        Console.WriteLine("Connected: {0}", client.Connected)
    
    End Sub 'ConnectAndCheck